lib/internal/errors.js

```diff
@@ -146,7 +126,27 @@ const maybeOverridePrepareStackTrace = (globalThis, error, trace) => {
     return MainContextError.prepareStackTrace(error, trace);
   }
 
-  return kNoOverride;
+  // Normal error formatting:
+  //
+  // Error: Message
+  //     at function (file)
+  //     at file
+  let errorString;
+  // Do not use primordials here: we intercept user code here.
+  const { toString } = error;
+  if (typeof toString === 'function' &&
+      toString !== MainContextObjectToString) {
+    errorString = FunctionPrototypeCall(toString, error);
+    if (errorString === '[Object object]' || typeof errorString !== 'string') {
```

Comment:
`Object.prototype.toString.call(new Error)` returns a different string btw:

```suggestion
    if (errorString === '[object Error]' || typeof errorString !== 'string') {
```

Reply:
I changed the check to a more generic one.